src/main/java/com/coreos/jetcd/EtcdClient.java

```diff
@@ -68,12 +70,15 @@ public EtcdClient(ManagedChannelBuilder<?> channelBuilder, EtcdClientBuilder bui
             mainFStub = setTokenForStub(mainFStub, token);
             mainStub = setTokenForStub(mainStub, token);
             clusterStub = setTokenForStub(clusterStub, token);
+            watchStub = setTokenForStub(watchStub, token);
         }
 
         this.kvClient = newKVClient(kvStub);
         this.authClient = newAuthClient(authStub);
         this.maintenanceClient = newMaintenanceClient(mainFStub, mainStub);
         this.clusterClient = newClusterClient(clusterStub);
+        this.watchClient = newWatchClient(watchStub);
+
```

Comment:
It is really needed to create so many objects bu default ? 
I would instead use Suppliers.memorize or any other way to have lazy object creation